### ISSUE TITLE
NVDA doesn't read the name of an element specified with aria-label or referenced with aria-labelledy if the element has dialog role

### ISSUE THREAD

**Issue by Otobelikethee** [nvaccess/nvda#9135]
demo: https://codesandbox.io/s/y2zj9mp741
### Steps to reproduce:
Open the demo online, and click the `Launch a modal` button.
### Actual behavior:
NVDA will announce: 
```
Header 
clickable 
button 
Close 
heading level 4
Header 
This is a modal 
button 
Close
```
It works like as if there are no `aria-labelledby` and `aria-describedby`.
### Expected behavior:
NVDA should announce:
```
Header, dialog 
This is a modal
```
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
2018.4
#### Windows version:
Windows 10 Pro Version 1703 Build 15063.674
#### Name and version of other software in use when reproducing the issue:
Firefox Version: 62.0.3 (32-bit)
#### Other information about your system:
Run Windows 10 Pro in the VirtualBox
#### Does the issue still occur after restarting your PC?
Yes
#### Have you tried any other versions of NVDA?
I have tried 2012.3.1 and it works as expected.

**Comment by Otobelikethee** [comment]
cc: @jcsteh

**Comment by jcsteh** [comment]
Not Firefox specific. Chrome exhibits the same behaviour.

That said:

> ### Actual behavior:
> 
> NVDA will announce:
> 
> ```
> *Header *
> ```

That there is the label. Without the label, it wouldn't have announced "Header" at this point.

> ### Expected behavior:
> 
> NVDA should announce:
> 
> ```
> Header, dialog 
> This is a modal
> ```

That is no longer the expected behaviour for dialogs within documents. Dialogs are now treated as browse mode documents due to overwhelming user demand and a change in ARIA 1.1. See #4493.

One thing I agree probably does need to be improved here is that "dialog" should be reported after the label.

**Comment by Otobelikethee** [comment]
@jcsteh Sorry, I don't understand why NVDA says all in the dialog when the dialog gets focused. Is it the behavior of **Browse Mode**? In my opinion, using `aria-describedby` should be more flexible to developers. 
I try to add a `application` role on the parent of element with `dialog` role and it works as expected. However, NVDA says all in the dialog when I tab into the content of dialog which has `document` role. I really don't want to see the "say all" behavior. Sad.


Also I have encountered another problem that the dialog is closed when I click the `Enter` key. At the same time, I can hear NVDA says "clickable close button..bla,bla...". Seems NVDA use a scanning cursor to read items in the dialog. And NVDA captures the command of `Enter` key and tells User Agent to dispatch a click on `close` button?

**Comment by Otobelikethee** [comment]
cc: @michaelDCurran , @leonardder , @josephsl .

**Comment by jcsteh** [comment]
> @jcsteh Sorry, I don't understand why NVDA says all in the dialog when the dialog gets focused. Is it the behavior of **Browse Mode**?

Yes. See #4493 for the background on why this change was made, but in short, the overwhelming opinion is that most dialogs on the web are more like documents rather than dialogs in the traditional desktop sense. That is, they tend to contain more content that the user needs to be able to browse, as opposed to just controls that the user tabs through. In a strict sense, I tend to agree with your stance on this, but the web as a whole does not and ARIA 1.1 was updated to reflect this. The reality is that what you want here would "break the web" for NVDA users. I'll stress again: this change was overwhelmingly desired by users.

**Comment by Otobelikethee** [comment]
@jcsteh Thanks for you further explanation. Could you please also help to explain the following another behavior based on current "says all" appearance in the dialog?

#### Another weird behavior: 
the dialog is closed when I click the Enter key. At the same time, I can hear NVDA says "clickable close button..bla,bla...". Seems NVDA use a scanning cursor to read items in the dialog. And NVDA captures the command of Enter key and tells User Agent to dispatch a click on close button?

**Comment by Adriani90** [comment]
@Otobelikethee NVDA performs the clickable action when you press enter on the close button in browse mode like it does in focus mode as well. This is expected behavior.

**Comment by Adriani90** [comment]
I am closing this issue in favor of #8620 to keep the focus on the announcement of "dialog" role when entering a dialog.
